nixos/modules/virtualisation/podman.nix

```diff
@@ -111,14 +184,50 @@ in
       };
 
       systemd.sockets.podman.wantedBy = [ "sockets.target" ];
+      systemd.sockets.podman.socketConfig.SocketGroup = "podman";
+
+      systemd.tmpfiles.packages = [
+        # The /run/podman rule interferes with our podman group, so we remove
+        # it and let the systemd socket logic take care of it.
+        (pkgs.runCommand "podman-tmpfiles-nixos" { package = cfg.package; } ''
+          mkdir -p $out/lib/tmpfiles.d/
+          grep -v 'D! /run/podman 0700 root root' \
+            <$package/lib/tmpfiles.d/podman.conf \
+            >$out/lib/tmpfiles.d/podman.conf
+        '') ];
+
+      systemd.tmpfiles.rules =
+        lib.optionals cfg.dockerSocket.enable [
+          "L! /run/docker.sock - - - - /run/podman/podman.sock"
+        ];
+
+      services.ghostunnel = lib.mkIf cfg.networkSocket.enable {
```

Comment:
Why is the `ghostunnel` stuff here and in the test, seems odd to add have the implementation of a specific proxy in the podman module. Can't it live somewhere else?

Reply:
> Why is the `ghostunnel` stuff here

Good point. I've moved it into a separate module and it's much cleaner now.
I've also added myself as a maintainer there.

> and in the test

I'm a hesitant to split the tests up further, because of the VM startup overhead. I'm not too concerned about Hydra (It's like 10 to 20 seconds perhaps) but about running the tests locally.
I did create a new test for the socket-based compatibility stuff, because the "core" podman tests in `nixosTests.podman` must work without the compatibility stuff.

> seems odd to add have the implementation of a specific proxy in the podman module. Can't it live somewhere else?

Having a module make an implementation choice isn't anything new. For instance, we have acme modules and options that are implemented by the `lego` client. In other words, it's ok for a module to be opinionated. I don't care about the choice of TLS server and I don't think swapping it out would cause any issues.